src/distrs/pois.jl

```diff
@@ -11,3 +11,12 @@ import .RFunctions:
     poisinvccdf,
     poisinvlogcdf,
     poisinvlogccdf
+
+# generic versions
+poispdf(λ::Real, x::Real) = exp(poislogpdf(λ, x))
+
+function poislogpdf(λ::T, x::T) where {T <: Real}
+    iszero(λ) ? (iszero(x) ? x : T(-Inf)) : xlogy(x, λ) - λ - lgamma(x + 1)
```

Comment:
I don't think the ternaries are needed here when you are using `xlogy`.
```julia
poislogpdf(λ::T, x::T) where {T <: Real} = xlogy(x, λ) - λ - lgamma(x + 1)
```
should be sufficient.

Reply:
Right.  Thanks.